### PR TITLE
Review troubleshooting OSDs

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jschmid@suse.de>

### CODE REVIEW

File: xml/bp_troubleshooting_osds.xml [hunk 1]
```diff
@@ -158,13 +164,15 @@
     <itemizedlist>
       <listitem>
         <para>
+<!--    Which configuration file are we talking about?     -->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml:167]: @jschmid1 I'd presume the ceph.conf at this point, which I'm not sure we'd want to encourage this as a course of action anymore?

File: xml/bp_troubleshooting_osds.xml [hunk 2]
```diff
@@ -230,6 +238,9 @@ osd.0 is down since epoch 23, last address 192.168.106.220:6800/11080
     <para>
       If there is a disk failure or other fault preventing <literal>ceph-osd</literal> from
       functioning or restarting, an error message should be present in its
+<!--        We still have log files for the ceph-osd daemon in this location but this is actually a bug :)-->
+<!--        We're currently discussing to move logging to journald entirely -->
+<!--        For a more comprehensive guide on logging see: https://docs.ceph.com/docs/master/cephadm/troubleshooting/#checking-cephadm-logs-->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml:243]: Added how to find the logs instead.

File: xml/bp_troubleshooting_osds.xml [hunk 3]
```diff
@@ -323,6 +335,7 @@ osd.0 is down since epoch 23, last address 192.168.106.220:6800/11080
     </para>
     <note>
       <para>
+<!--          Currently, partition based db/wals are not supported.-->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml:338]: Removed this note

File: xml/bp_troubleshooting_osds.xml [hunk 4]
```diff
@@ -310,6 +321,7 @@ osd.0 is down since epoch 23, last address 192.168.106.220:6800/11080
   <sect2 xml:id="troubleshooting-drive-config">
     <title>Drive Configuration</title>
     <para>
+<!--       Nowadays the exception is fast SSDs and NVMEs -->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml:324]: Unsure what needs to be changed here? @jschmid1

File: xml/bp_troubleshooting_osds.xml [hunk 5]
```diff
@@ -712,6 +725,7 @@ osd.0 is down since epoch 23, last address 192.168.106.220:6800/11080
    script.
   </para>
   <para>
+<!--      This isn't really a thing anymore in newer releases.-->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml:728]: Removed

File: xml/bp_troubleshooting_osds.xml [hunk 6]
```diff
@@ -759,13 +773,29 @@ osd.0 is down since epoch 23, last address 192.168.106.220:6800/11080
    you may check if the disk where the OSD is located is mounted:
   </para>
 <screen>&prompt.root;lsblk -f
+<!--    the output has changed a bit:-->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml:776]: @jschmid1 could you provide an updated output?

File: xml/bp_troubleshooting_osds.xml [hunk 7]
```diff
@@ -759,13 +773,29 @@ osd.0 is down since epoch 23, last address 192.168.106.220:6800/11080
...
+<!--dev-box-1:~ #-->
+
  [...]
  vdb
  ├─vdb1               /var/lib/ceph/osd/ceph-2
  └─vdb2
 </screen>
   <para>
    You can track the reason why the OSD is down by inspecting its log file
+<!--      Here again, the log location is likely to change in the near future -->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml:798]: Added a cross-ref to the log location

File: xml/bp_troubleshooting_osds.xml [hunk 8]
```diff
@@ -127,6 +127,11 @@
      Once the cluster is set to <option>noout</option>, you can begin stopping
      the OSDs within the failure domain that requires maintenance work:
    </para>
+<!--     something is missing here. <stop> without a prepended command won't work -->
+<!--     I'd suggest we use cephadms `daemon` command for stopping/starting a deamon.-->
+<!--     The `ceph osd set noout` can remain as is.-->
+<!--     The command would be: `ceph orch daemon start|stop osd.$ID`-->
+<!--     example output: `Scheduled to stop|start osd.$ID on host 'example_host'`-->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml]: Updated

File: xml/bp_troubleshooting_osds.xml [hunk 9]
```diff
@@ -137,6 +142,7 @@
    <para>
      Once you have completed your maintenance, restart the OSDs:
    </para>
+<!--     Same remarks with <start> -->
```
Comment (asettle) [inline xml/bp_troubleshooting_osds.xml]: Ditto

File: xml/bp_troubleshooting_osds.xml (hunk 1 above)
Comment (jschmid1) [reply xml/bp_troubleshooting_osds.xml:167]: I *assume* we talked about salt files here. In cephadm times we don't explicitly list devices anymore so maybe whole section can disappear and is being replaced by the drivegroup/osdspec troubleshooting guide (which doesn't really exist, yet :))

I'd get rid of this for now..

File: xml/bp_troubleshooting_osds.xml (hunk 4 above)
Comment (jschmid1) [reply xml/bp_troubleshooting_osds.xml:320]: Imo, this is not troubleshooting but general guidance and I'm pretty sure we have paragraphs about that.

File: xml/bp_troubleshooting_osds.xml (hunk 6 above)
Comment (jschmid1) [reply xml/bp_troubleshooting_osds.xml:750]: uch, I actually pasted it below this comment.. The formatting sucks though. Re-pasting:

```

dev-box-1:~ # lsblk -f
NAME                                                                                                  FSTYPE      LABEL UUID                                   FSAVAIL FSUSE% MOUNTPOINT
vda
├─vda1
├─vda2                                                                                                vfat        EFI   7BDD-40C3                                18.8M     6% /boot/efi
└─vda3                                                                                                ext4        ROOT  144d3eec-c193-4793-b6a9-1ac295259f4b     36.7G     5% /
vdb                                                                                                   LVM2_member       Fj8nlY-Dnmm-8Y0w-tJrO-8rAe-SUTH-lA2Ux2
└─ceph--6dc6f71f--ff02--4316--b145--c7804d5fb2f4-osd--block--2cb16f65--c87d--405d--a913--4e4ea6037ca1
vdc                                                                                                   LVM2_member       0101m1-mc83-K8ce-j4Dv-yxUO-5KPj-6FpuSu
└─ceph--1082eac0--4478--48cf--b8ab--4d3a62ca1c27-osd--data--185c9dd6--ce28--4b98--a9b6--9d5b1f444a4f
vdd                                                                                                   LVM2_member       wqE8sC-w5mx-J9t1-FoM1-vIRC-0xxq-5FPyuL
└─ceph--9e71d81a--e394--49b7--bef9--b639083be779-osd--data--eb94f8e6--af6d--4ef6--911b--7f44f7484c85


```

File: xml/bp_troubleshooting_osds.xml (hunk 1 above)
Comment (asettle) [reply xml/bp_troubleshooting_osds.xml:167]: Removed :)

File: xml/bp_troubleshooting_osds.xml (hunk 4 above)
Comment (asettle) [reply xml/bp_troubleshooting_osds.xml:320]: Whole section removed :D

File: xml/bp_troubleshooting_osds.xml (hunk 6 above)
Comment (asettle) [reply xml/bp_troubleshooting_osds.xml:750]: I see! That did format oddly. Thanks!

**Review by asettle — APPROVED** [review]

**Pull request merged**